### ISSUE TITLE
Invoking runner.cloud.action via reactor sls fails

### ISSUE THREAD

**Issue by DaveOHenry** [saltstack/salt#38839]
### Description of Issue/Question
We are using EC2 and I tried to invoke a cloud runner action via the event system, which fails with an exception:
```
salt-run state.event pretty=true
```
```
my_test_event	{
    "_stamp": "2017-01-20T08:29:41.956660", 
    "cmd": "_minion_event", 
    "data": {
        "__pub_fun": "event.send", 
        "__pub_jid": "20170120092941938085", 
        "__pub_pid": 14303, 
        "__pub_tgt": "salt-call"
    }, 
    "id": "my_saltmaster", 
    "tag": "my_test_event"
}
salt/job/20170120092941973697/ret/my_saltmaster	{
    "_stamp": "2017-01-20T08:29:41.976136", 
    "arg": [
        "my_test_event"
    ], 
    "cmd": "_return", 
    "fun": "event.send", 
    "fun_args": [
        "my_test_event"
    ], 
    "id": "my_saltmaster", 
    "jid": "20170120092941973697", 
    "retcode": 0, 
    "return": true, 
    "tgt": "my_saltmaster", 
    "tgt_type": "glob"
}
salt/run/20170120092942802583/new	{
    "_stamp": "2017-01-20T08:29:42.806233", 
    "fun": "runner.cloud.action", 
    "fun_args": [
        {
            "__id__": "cloudrunner", 
            "__sls__": "/srv/reactor/my_test_reactor.sls", 
            "cloudmap": null, 
            "func": "disable_term_protect", 
            "instance": "my_minion", 
            "instances": null, 
            "name": "cloudrunner", 
            "order": 1, 
            "provider": null, 
            "state": "runner"
        }
    ], 
    "jid": "20170120092942802583", 
    "user": "Reactor"
}
salt/run/20170120092942802583/ret	{
    "_stamp": "2017-01-20T08:29:43.049610", 
    "fun": "runner.cloud.action", 
    "fun_args": [
        {
            "__id__": "cloudrunner", 
            "__sls__": "/srv/reactor/my_test_reactor.sls", 
            "cloudmap": null, 
            "func": "disable_term_protect", 
            "instance": "my_minion", 
            "instances": null, 
            "name": "cloudrunner", 
            "order": 1, 
            "provider": null, 
            "state": "runner"
        }
    ], 
    "jid": "20170120092942802583", 
    "return": "Exception occurred in runner cloud.action: Traceback (most recent call last):\n  File \"/usr/lib/python2.7/dist-packages/salt/client/mixins.py\", line 395, in _low\n    data['return'] = self.functions[fun](*args, **kwargs)\n  File \"/usr/lib/python2.7/dist-packages/salt/runners/cloud.py\", line 154, in action\n    info = client.action(func, cloudmap, instances, provider, instance, **_filter_kwargs(kwargs))\nTypeError: action() got an unexpected keyword argument 'state'\n", 
    "success": false, 
    "user": "Reactor"
}
```

Manually executing the action works as expected:
```
salt-run cloud.action func=enable_term_protect instance=my_minion
```

Triggering cloud.profile (for example) via the reactor works fine, too: The instance is created without a problem. I think there is something strange happening inside cloud.action.

### Setup
/etc/salt/master.d/reactor.conf
```
reactor:
  - 'my_test_event':
    - /srv/reactor/my_test_reactor.sls
```

/srv/salt/reactor/my_test_reactor.sls
```
cloudrunner:
  runner.cloud.action:
    - func: enable_term_protect
    - instance: my_minion
```

### Steps to Reproduce Issue
```
salt-call event.send 'my_test_event'
```

### Versions Report
```
Salt Version:
           Salt: 2016.11.1
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.5.1
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 0.9.1
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.6 (default, Oct 26 2016, 20:30:19)
   python-gnupg: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.4.0
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
 
System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.19.0-73-generic
         system: Linux
        version: Ubuntu 14.04 trusty
```

**Comment by Ch3LL** [comment]
@DaveOHenry heres the stack trace prettied up a bit to make it easier to read:

```
Traceback (most recent call last):\n  
 File \"/usr/lib/python2.7/dist-packages/salt/client/mixins.py\", line 395, in _low\n    
 data['return'] = self.functions[fun](*args, **kwargs)\n  
 File \"/usr/lib/python2.7/dist-packages/salt/runners/cloud.py\", line 154, in action\n
 info = client.action(func, cloudmap, instances, provider, instance, **_filter_kwargs(kwargs))\n
 TypeError: action() got an unexpected keyword argument 'state'\n", 
```

When attempting this myself this is what I see in the event as well:

```
    "fun": "runner.cloud.action", 
    "fun_args": [
        {
            "__id__": "cloudrunner", 
            "__sls__": "/srv/reactor/my_test_reactor.sls", 
            "cloudmap": null, 
            "func": "enable_term_protect", 
            "instance": "my_minion", 
            "instances": null, 
            "name": "cloudrunner", 
            "order": 1, 
            "provider": null, 
            "state": "runner"
        }
```

I'm guessing its passing that state kwarg into the action call and thats where the issue lies. The action call looks for `kwargs=None`.  Looks like we need to fix this up thanks.

**Comment by vutny** [comment]
Bump. This issue is still present in 2016.11.6.

**Comment by vutny** [comment]
@Ch3LL I did an attempt to fix this in PR  #42291. Please take a look.

**Comment by vutny** [comment]
Finally, the fix was merged 🍾 FYI @Ch3LL @rallytime 

The workaround until next release is to use orchestration function:
```yaml
# /srv/salt/reactor/my_test_reactor.sls

cloudrunner-reactor:
  runner.state.orchestrate:
    - mods: orch.cloud
    - kwarg:
        pillar:
          event_data: my_minion
```

```yaml
# /srv/salt/orch/cloud.sls

cloudrunner-orch:
  runner.cloud.action:
    - func: enable_term_protect
    - instance: pillar['event_data']
```

**Comment by rallytime** [comment]
Thanks for fixing this and for posting the workaround @vutny.

This fix will be available in 2016.11.8.
